examples/versions.tf

```diff
@@ -11,7 +11,7 @@ terraform {
   required_providers {
     gitlab = {
       source  = "gitlabhq/gitlab"
-      version = "3.11.1"
+      version = "~> 16.3"
```

Comment:
I can't see relation between terraform version upgrade and gitlab provider. Could you please share reason to bump both at the same time?

Reply:
I don't believe it is _necessary_ to bump both at the same time.
I guess that you're at least as aware of what's going on than I am, but just in case, some points that I learned recently are interesting to me:
* Since December 2022, the gitlabhq/gitlab provider has moved to https://gitlab.com/gitlab-org/terraform-provider-gitlab and is being versioned in sync with gitlab itself for simplicity (currently 16.x).
* For versions `>15.7,<16`, the only breaking changes, listed in the [upgrade guide](https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs/guides/version-15.7-upgrade) are that it requires terraform >=1.0, which these modules already require, and the `sensitive` protection for the token.
* For 16+, there are [rather more](https://registry.terraform.io/providers/gitlabhq/gitlab/latest/docs/guides/version-16.0-upgrade), so perhaps you'll want to pin to <16?
* There has been masses of development (both feature and bugfix) in the provider since 3.x, including in areas that affect your modules.

If you want to stay with an older version because your users want or need to compatible with older versions of the gitlab platform instances (i.e. not gitlab.com), that's understandable.

Since Terraform allows for easy pinning of provider/module version, one possibility would be to have branches of this codebase that are named for the major version of the gitlab provider that they use?

Reply:
So if there are no dependencies could you please revert changes for gitlab provider and keep this PR specific to terraform version upgrade. Then open a new PR upgrading gitlab provider to 15.x as it would force us to make major version release of this module.

Thank you for your contribution!

Reply:
okay, updated this PR and added https://github.com/EvoltDev/terraform-gitlab-gitlab/pull/16